src/odemis/acq/stream/_sync.py

```diff
@@ -1140,46 +1143,55 @@ def _runAcquisitionEbeam(self, future):
     def _integrateImages(self, n):
         """
         Integrate/sum the last n acquired images in self._acq_data and replace by the
-        integrated image. If values overflow, clip the image.
+        integrated image. If values overflow, clip the image. If number of images = 1,
+        no image integration necessary.
         :param n: (int) Number of images that need to be integrated (summed).
         """
-        # TODO is there an sem_idx or similar? How to handle if more than 1 sem det -> for loop?
-        # get the SEM images corresponding to the last acquisition and average
-        data_sem = numpy.mean(self._acq_data[0][-n:], axis=0)
-        # TODO anything need to consider regarding type?
+        # check if we need to integrate images
+        if not n == 1:
```

Comment:
Invert the test case, and immediately return if n == 1. This way, you gain one indentation level ;-)